modules/renter/hostdb/hostdb.go

```diff
@@ -39,6 +40,110 @@ var (
 	errHostNotFoundInTree = errors.New("host not found in hosttree")
 )
 
+// blockedDomains manages a list of blocked domains
+type blockedDomains struct {
+	domains map[string]struct{}
+	mu      sync.Mutex
+}
+
+// newBlockedDomains initializes a new blockedDomains
+func newBlockedDomains(domains []string) *blockedDomains {
+	domainMap := make(map[string]struct{})
+	for _, domain := range domains {
+		domainMap[domain] = struct{}{}
+	}
+	return &blockedDomains{
+		domains: domainMap,
+	}
+}
+
+// managedAddDomains adds domains to the map of blocked domains
+func (bd *blockedDomains) managedAddDomains(domains []string) {
+	bd.mu.Lock()
+	defer bd.mu.Unlock()
+	for _, domain := range domains {
+		bd.domains[domain] = struct{}{}
+	}
+}
+
+// managedBlockedDomains returns a list of the blocked domains
+func (bd *blockedDomains) managedBlockedDomains() []string {
+	bd.mu.Lock()
+	defer bd.mu.Unlock()
+	var domains []string
+	for domain := range bd.domains {
+		domains = append(domains, domain)
+	}
+
+	return domains
+}
+
+// managedRemoveDomains removes domains from the map of blocked domains
+func (bd *blockedDomains) managedRemoveDomains(domains []string) {
+	bd.mu.Lock()
+	defer bd.mu.Unlock()
+	for _, domain := range domains {
+		delete(bd.domains, domain)
+	}
+}
+
+// managedIsBlocked checks to see if the domain is blocked
+func (bd *blockedDomains) managedIsBlocked(addr modules.NetAddress) bool {
```

Comment:
It'd be useful to also resolve the DNS host and check against its A and AAAA records.